### DISCUSSION TITLE
现有docker镜像不支持

### DISCUSSION THREAD

**Discussion by zhangyifei1** [PaddlePaddle/PaddleSpeech#575] [General]
我试了几个版本，docker镜像系统版本都是ubuntu16.04。然而这个版本的系统，安装resampy这个包的时候不支持0.2.2，导致整个项目都跑不动。
建议改动方法：将docker环境升级至ubuntu18.04，这样就可以装上去了。
既然要开源，就应该把服务搞好，百度的技术没得说，怎么把拿出去的东西搞的这么难使用呢？头大好几天了，忍不住吐槽下。请开发人员重视，上线一个能成功执行的docker环境。谢谢

**Comment by zhangyifei1** [comment]
这个docker里面默认python3还是3.5的

**Reply by zh794390558** [reply]
默认的是3.5， tools 下make下会安装venv环境python3.7.

**Comment by zhangyifei1** [comment]
docker环境下执行setep.sh



λ bb2459ef9ff8 /DeepSpeech {develop} sh setup.sh 
setup.sh: 3: setup.sh: source: not found
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libboost-dev is already the newest version (1.58.0.1ubuntu1).
libflac-dev is already the newest version (1.3.1-4).
libogg-dev is already the newest version (1.3.2-1).
pkg-config is already the newest version (0.29.1-0ubuntu1).
python3-dev is already the newest version (3.5.1-3).
swig is already the newest version (3.0.8-0ubuntu3).
libvorbis-dev is already the newest version (1.3.5-3ubuntu0.2).
0 upgraded, 0 newly installed, 0 to remove and 6 not upgraded.
/usr/local/python3.5.1/lib/python3.5/site-packages/OpenSSL/crypto.py:14: CryptographyDeprecationWarning: Python 3.5 support will be dropped in the next release of cryptography. Please upgrade your Python.
  from cryptography import utils, x509
Requirement already satisfied: scipy==1.2.1 in /usr/local/python3.5.1/lib/python3.5/site-packages (from -r requirements.txt (line 1)) (1.2.1)
Collecting resampy==0.2.2 (from -r requirements.txt (line 2))
Collecting SoundFile==0.9.0.post1 (from -r requirements.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/bc/16/8f35e0e96b6fb94ee63313409fc8bfccc29c1c5c89f6ab83a5fdfc4c68ac/SoundFile-0.9.0.post1-py2.py3-none-any.whl
Collecting python_speech_features (from -r requirements.txt (line 4))
Collecting tensorboardX (from -r requirements.txt (line 5))
  Using cached https://files.pythonhosted.org/packages/af/0c/4f41bcd45db376e6fe5c619c01100e9b7531c55791b7244815bac6eac32c/tensorboardX-2.1-py2.py3-none-any.whl
Collecting yacs (from -r requirements.txt (line 6))
  Using cached https://files.pythonhosted.org/packages/38/4f/fe9a4d472aa867878ce3bb7efb16654c5d63672b86dc0e6e953a67018433/yacs-0.1.8-py3-none-any.whl
Collecting typeguard (from -r requirements.txt (line 7))
  Using cached https://files.pythonhosted.org/packages/de/60/8b67e520d3d45bb385322cd0603ad055cebcd4006dc46a74d4f32a3792a9/typeguard-2.2.2-py3-none-any.whl
Requirement already satisfied: numpy>=1.8.2 in /usr/local/python3.5.1/lib/python3.5/site-packages (from scipy==1.2.1->-r requirements.txt (line 1)) (1.18.5)
Collecting numba>=0.32 (from resampy==0.2.2->-r requirements.txt (line 2))
  Using cached https://files.pythonhosted.org/packages/05/c9/ad7e8968197028d4d55fe5f03c54d0f48cf7e975aca2d2bdccf9610d511b/numba-0.47.0-cp35-cp35m-manylinux1_x86_64.whl
Requirement already satisfied: six>=1.3 in /usr/local/python3.5.1/lib/python3.5/site-packages (from resampy==0.2.2->-r requirements.txt (line 2)) (1.15.0)
Requirement already satisfied: cffi>=0.6 in /usr/local/python3.5.1/lib/python3.5/site-packages (from SoundFile==0.9.0.post1->-r requirements.txt (line 3)) (1.14.4)
Requirement already satisfied: protobuf>=3.8.0 in /usr/local/python3.5.1/lib/python3.5/site-packages (from tensorboardX->-r requirements.txt (line 5)) (3.14.0)
Requirement already satisfied: PyYAML in /usr/local/python3.5.1/lib/python3.5/site-packages (from yacs->-r requirements.txt (line 6)) (5.3.1)
Requirement already satisfied: setuptools in /usr/local/python3.5.1/lib/python3.5/site-packages (from numba>=0.32->resampy==0.2.2->-r requirements.txt (line 2)) (50.3.2)
Collecting llvmlite>=0.31.0dev0 (from numba>=0.32->resampy==0.2.2->-r requirements.txt (line 2))
  Using cached https://files.pythonhosted.org/packages/50/cc/04526507e80d546be5688ce0246e40277b61e7949c3347c6609b6a4154cf/llvmlite-0.32.1.tar.gz
Requirement already satisfied: pycparser in /usr/local/python3.5.1/lib/python3.5/site-packages (from cffi>=0.6->SoundFile==0.9.0.post1->-r requirements.txt (line 3)) (2.20)
Building wheels for collected packages: llvmlite
  Running setup.py bdist_wheel for llvmlite ... error
  Complete output from command /usr/local/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-zhv5pq1g/llvmlite/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-te_ftmjw --python-tag cp35:
  running bdist_wheel
  /usr/local/bin/python3 /tmp/pip-install-zhv5pq1g/llvmlite/ffi/build.py
  LLVM version... 3.8.0
  
  Traceback (most recent call last):
    File "/tmp/pip-install-zhv5pq1g/llvmlite/ffi/build.py", line 192, in <module>
      main()
    File "/tmp/pip-install-zhv5pq1g/llvmlite/ffi/build.py", line 182, in main
      main_posix('linux', '.so')
    File "/tmp/pip-install-zhv5pq1g/llvmlite/ffi/build.py", line 144, in main_posix
      raise RuntimeError(msg)
  RuntimeError: Building llvmlite requires LLVM 7.0.x, 7.1.x or 8.0.x, got '3.8.0'. Be sure to set LLVM_CONFIG to the right executable path.
  Read the documentation at http://llvmlite.pydata.org/ for more information about building llvmlite.
  
  error: command '/usr/local/bin/python3' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for llvmlite
  Running setup.py clean for llvmlite
Failed to build llvmlite
Installing collected packages: llvmlite, numba, resampy, SoundFile, python-speech-features, tensorboardX, yacs, typeguard
  Running setup.py install for llvmlite ... error
    Complete output from command /usr/local/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-zhv5pq1g/llvmlite/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-vs4mb33s/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    got version from file /tmp/pip-install-zhv5pq1g/llvmlite/llvmlite/_version.py {'version': '0.32.1', 'full': 'aa11b129c0b55973067422397821ae6d44fa5e70'}
    running build_ext
    /usr/local/bin/python3 /tmp/pip-install-zhv5pq1g/llvmlite/ffi/build.py
    LLVM version... 3.8.0
    
    Traceback (most recent call last):
      File "/tmp/pip-install-zhv5pq1g/llvmlite/ffi/build.py", line 192, in <module>
        main()
      File "/tmp/pip-install-zhv5pq1g/llvmlite/ffi/build.py", line 182, in main
        main_posix('linux', '.so')
      File "/tmp/pip-install-zhv5pq1g/llvmlite/ffi/build.py", line 144, in main_posix
        raise RuntimeError(msg)
    RuntimeError: Building llvmlite requires LLVM 7.0.x, 7.1.x or 8.0.x, got '3.8.0'. Be sure to set LLVM_CONFIG to the right executable path.
    Read the documentation at http://llvmlite.pydata.org/ for more information about building llvmlite.
    
    error: command '/usr/local/bin/python3' failed with exit status 1

**Reply by zh794390558** [reply]
https://github.com/PaddlePaddle/DeepSpeech/blob/develop/docs/install.md#setup 看下这个文档里setup的二三步。

**Comment by leijue222** [comment]
I don't how to test pre-trained model after run the Docker image.
The docs files seem lost this step.

**Reply by leijue222** [reply]
很想试一下Mandarin LM Large的效果，目前试了Espnet, mozilla/DeepSpeech两个版本准确率都不尽人意。

**Reply by zh794390558** [reply]
2.x没有放出来ASR的model，需要子集训练下。egs里的指标应该用的不是large的LM，可以试下large后能调到多少。transformer的模型还在开发中，后续回放出来，之后也会放和上线相关的代码，可以关注下。
